app/src/main/java/tech/relaycorp/gateway/background/ConnectionState.kt

```diff
@@ -0,0 +1,9 @@
+package tech.relaycorp.gateway.background
+
+sealed class ConnectionState {
+    object Disconnected : ConnectionState()
+    object InternetWithoutRelaynet : ConnectionState()
+    object InternetAndRelaynet : ConnectionState()
```

Comment:
I'd suggest to replace `Relaynet` with `PublicGateway` for consistency with the rest of the code in the PR and to be more precise.